pkg/collector/corechecks/snmp/internal/profile/config_profile.go

```diff
@@ -9,10 +9,54 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/networkdevice/profile/profiledefinition"
 )
 
-// ProfileConfigMap represent a map of ProfileConfig
+// Provider is an interface that provides profiles by name
+type Provider interface {
+	// HasProfile returns true if and only if we have a profile by this name.
+	HasProfile(profileName string) bool
+	// GetProfile returns the profile with this name, or nil if there isn't one.
+	GetProfile(profileName string) *ProfileConfig
+	// GetProfileNameForSysObjectID returns the best matching profile for this sysObjectID, or nil if there isn't one.
+	GetProfileNameForSysObjectID(sysObjectID string) (string, error)
+	// GetAllProfiles returns a map from names to profiles.
+	GetAllProfiles() ProfileConfigMap
+}
+
+// staticProvider is a static implementation of Provider
+type staticProvider struct {
+	configMap ProfileConfigMap
+}
+
+func (s *staticProvider) GetProfile(name string) *ProfileConfig {
+	if profile, ok := s.configMap[name]; ok {
+		return &profile
+	}
+	return nil
+}
+
+func (s *staticProvider) HasProfile(profileName string) bool {
+	_, ok := s.configMap[profileName]
+	return ok
+}
+
+func (s *staticProvider) GetProfileNameForSysObjectID(sysObjectID string) (string, error) {
+	return getProfileForSysObjectID(s.configMap, sysObjectID)
+}
+
+func (s *staticProvider) GetAllProfiles() ProfileConfigMap {
+	return s.configMap
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏There's a slight risk here (and in GetProfile) that the caller can modify the content of the profiles. Did you see it and would you think it's worth to mitigate it?

Reply:
For GetAllProfiles it seemed excessive to clone them all; now that we don't have that anymore, it's just `GetProfile` that has this risk. I didn't want to copy them because I thought it would be wasteful to copy a profile every time we access it, but maybe that's just me thinking too much like a Python programmer - probably the time needed to duplicate a profile is tiny enough that we can just go ahead and do it every time. Do you think it would be better to just duplicate the profile every time `GetProfile()` is called?